content/develop/tutorials/databases/snowflake.md

```diff
@@ -7,151 +7,220 @@ slug: /develop/tutorials/databases/snowflake
+This guide explains how to securely access a Snowflake database from Streamlit. It uses [st.connection](/develop/api-reference/connections/st.connection), the [Snowpark library](https://docs.snowflake.com/en/developer-guide/snowpark/python/index) and Streamlit's [Secrets management](/develop/concepts/connections/secrets-management).
+
+### Prerequisites
+
+- The following packages must be installed in your Python environment:
+
+  ```txt
+  streamlit>=1.28
+  snowflake-snowpark-python>=0.9.0
+  snowflake-connector-python>=2.8.0
+  ```
+
+    <Note>
+        Use the correct version of Python required by `snowflake-snowpark-python`. For example, if you use `snowflake-snowpark-python==1.23.0` you must use Python version \>=3.8, \<3.12.
+    </Note>
+
+- You must have a Snowflake account. To create a trial account, see the [tutorial](/get-started/installation/streamlit-in-snowflake) in _Get started_.
+- You should have a basic understanding of [`st.connection`](/develop/api-reference/connections/st.connection) and [Secrets management](/develop/concepts/connections/secrets-management).
 
 ## Create a Snowflake database
 
-<Note>
+If you already have a database that you want to use, you can [skip to the next step](#add-connection-parameters-to-your-local-app-secrets).
 
-If you already have a database that you want to use, feel free to [skip to the next step](#add-username-and-password-to-your-local-app-secrets).
+1. Sign in to your Snowflake account at [https://app.snowflake.com](https://app.snowflake.com).
+1. In the left navigation, click "**Projects**," then click "**Worksheets**."
+1. To create a new worksheet, in the upper-right corner, click the plus icon (<i style={{ verticalAlign: "-.25em" }} className={{ class: "material-icons-sharp" }}>add</i>).
 
-</Note>
+   You can use a worksheet to quickly and conveniently execute SQL statements. This is a great way to learn about and experiment with SQL in a trial account.
 
-First, [sign up for Snowflake](https://signup.snowflake.com/) and log into the [Snowflake web interface](https://docs.snowflake.com/en/user-guide/connecting.html#logging-in-using-the-web-interface) (note down your username, password, and [account identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html)!):
+1. Optional: To rename your worksheet, in the upper-left corner, hover over the tab with your worksheet name, and click the overflow menu icon (<i style={{ verticalAlign: "-.25em" }} className={{ class: "material-icons-sharp" }}>more_vert</i>). Select "**Rename**", enter a new worksheet name (e.g. "Scratchwork"), and press "**Enter**".
```

Comment:
instead of "Do X, do Y, and do Z," the standard phraseology is "Do X, do Y,, and **then** do Z." 

Same standard for only two actions: "Do X, and then do Y." 

I got my Xs and Ys confused, which is worrisome.